monkey/infection_monkey/control.py

```diff
@@ -42,13 +45,22 @@ def wakeup(parent=None, has_internet_access=None):
         if has_internet_access is None:
             has_internet_access = check_internet_access(WormConfiguration.internet_services)
 
+        user = "Unknown standard user"
+        try:
+            if os.getuid():  # won't throw an exception if it's linux
+                user = os.popen('whoami').read()[:-1]  # get the username
+        except AttributeError:
+            if user_token_is_admin(0):
+                user = "Admin"
+
         monkey = {'guid': GUID,
                   'hostname': hostname,
                   'ip_addresses': local_ips(),
                   'description': " ".join(platform.uname()),
                   'internet_access': has_internet_access,
                   'config': WormConfiguration.as_dict(),
-                  'parent': parent}
+                  'parent': parent,
+                  'root': user}
```

Comment:
what is 'root'? a boolean or a username string? If it's username than maybe it should be called 'username_monkey_runs_as' or something similar

Reply:
root is defined as a stringfield(), It initially bool, but then modified to hold the username. I'll rename it :)